optics-core/src/Optics/Internal/Optic/Types.hs

```diff
@@ -86,12 +86,12 @@ type family Constraints (k :: OpticKind) (p :: * -> * -> * -> *) :: Constraint w
   Constraints 'A_Prism            p = Choice p
   Constraints 'A_PrismaticGetter  p = Cochoice p
   Constraints 'An_AffineTraversal p = (Strong p, Choice p)
-  Constraints 'A_Traversal        p = Traversing p
-  Constraints 'An_IxTraversal     p = TraversingWithIndex p
-  Constraints 'A_Setter           p = Mapping p
-  Constraints 'An_IxSetter        p = p ~ IxFunArrow
+  Constraints 'A_Traversal        p = IxTraversing p
+  Constraints 'An_IxTraversal     p = IxTraversing p
```

Comment:
I feel uneasy about this.

Do we still have

```haskell
traverseOf traversed = traverse
traverseOf (traversed % traversed) = traverse . traverse
```

(I think `inspection-testing`)

But if this works, then #41 can be tweaked even further into nicer form indeed.

Reply:
Yeah, nothing changes with regard to operations on regular optics, the extra constraint is just so that indexed variants can be used as regular ones (as Ix* instances for Forget, Star and FunArrow simply ignore indices).

Reply:
Ah, you mean code generation? I didn't check.